backend/tracim_backend/lib/core/plugins.py

```diff
@@ -0,0 +1,106 @@
+import collections
+import importlib
+import pkgutil
+import sys
+import types
+import typing
+
+import pluggy
+
+from tracim_backend.config import CFG
+from tracim_backend.lib.utils.utils import find_all_submodule_path
+
+EVENT_NAMESPACE = "tracim_event"
+HOOKSPEC_FILES_SUFFIX = "hookspec"
+hookspec = pluggy.HookspecMarker(EVENT_NAMESPACE)
+hookimpl = pluggy.HookimplMarker(EVENT_NAMESPACE)
+
+
+class TracimPluginManager(object):
+    def __init__(self, prefix: str):
+        self.prefix = prefix
+        self.loaded_plugin = False
+        self.plugins = {}
+        self.event_dispatcher = self._setup_pluggy_event_dispatcher()
+
+    @classmethod
+    def get_all_hookspec_module_path(cls):
+        hookspec_module_paths = []
+        # INFO - G.M - 2019-11-28 - import root module,
+        # currently "tracim_backend"
+        root_module_name = cls.__module__.split(".")[0]
+        root_module = importlib.import_module(root_module_name)
+        # INFO - G.M - 2019-11-28 - get all submodules recursively,
+        # find those with with hookspec suffix, import them and add them
+        # to valid hookspec
+        for module_path in find_all_submodule_path(root_module):
+            if module_path.endswith(HOOKSPEC_FILES_SUFFIX):
+                hookspec_module_paths.append(module_path)
+        return hookspec_module_paths
+
+    def _load_spec(self, event_dispatcher: pluggy.PluginManager) -> None:
+        """
+        Load all hookspec modules
+        """
+
+        # INFO - G.M - 2019-11-28 - get all submodules recursively,
+        # find those with with hookspec suffix, import them and add them
+        # to valid hookspec
+        for hookspec_module_path in self.get_all_hookspec_module_path():
+            module = importlib.import_module(hookspec_module_path)
+            event_dispatcher.add_hookspecs(module)
+
+    def _setup_pluggy_event_dispatcher(self) -> pluggy.PluginManager:
```

Comment:
Prefer `create` naming for creation purpose

Reply:
done.